From 1d596ba8cec0b9ed7b3b810dc0c511ebb9b7e1d5 Mon Sep 17 00:00:00 2001
From: Gustavo Solaira <gustavos@qti.qualcomm.com>
Date: Wed, 18 Feb 2015 18:07:30 -0200
Subject: [PATCH] ARM: mach-msm: Enable independent SSR by default for
 Constitution

Enable independent SSR by default for the Constitution board,
this will enable each subsystem to crash independently and not
crash the SoC. Critical failures on APPS and RPM will still cause
a full device reset.
---
 arch/arm/mach-msm/subsystem_restart.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-msm/subsystem_restart.c b/arch/arm/mach-msm/subsystem_restart.c
index 747276c..358cb4d 100644
--- a/arch/arm/mach-msm/subsystem_restart.c
+++ b/arch/arm/mach-msm/subsystem_restart.c
@@ -638,7 +638,7 @@ static int __init ssr_init_soc_restart_orders(void)
 
 static int __init subsys_restart_init(void)
 {
-	restart_level = RESET_SOC;
+	restart_level = RESET_SUBSYS_INDEPENDENT;
 
 	ssr_wq = alloc_workqueue("ssr_wq", WQ_CPU_INTENSIVE, 0);
 	if (!ssr_wq)
-- 
1.7.9.5

